tests/test_file_processing.py

```diff
@@ -913,6 +913,36 @@ def test_where_positive_result_does_not_align_with_ct_channel_results(config):
         assert centre_file.logging_collection.get_count_of_all_errors_and_criticals() == 1
 
 
+def test_remove_bom(centre_file):
+    with StringIO() as fake_csv:
+        # construct a bytes object containing a byte order mark (BOM)
+        header_with_bom = b"\xef\xbb\xbfRoot Sample ID"
+        bom_as_utf8_string = header_with_bom.decode("utf-8")
+
+        fake_csv.write(f"{bom_as_utf8_string},RNA ID\n")
+        fake_csv.write("1,RNA_0043_\n")
+        fake_csv.seek(0)
+
+        csv_to_test_reader = DictReader(fake_csv)
+        assert csv_to_test_reader.fieldnames != ["Root Sample ID", "RNA ID"]
```

Comment:
I think you're just checking the standard library here.

Reply:
I'm doing this check here to make sure that we initially don't get these column headers, so we know that when we reach the second assertion on line 930, we know for sure that the remove_bom method is responsible for the change to the correct column headers.

On top of that, I found checking for the presence of a BOM a bit slippery! Some standard python functions seemed to auto-remove it, so it looked like it wasn't there when it was (specifically, when they were in a list, you could see the BOM, but when you selected an item from the list via indexing, you could no longer see the BOM). So it's good to have an assertion in there proving the BOM is present, before we prove it's not present.

Similar answer to line 940 (minus the BOM bit). 

Does that make sense or have I misunderstood your comment?

Reply:
OK, yes thank you, that does make sense!